examples/in_gdummy/in_gdummy.go

```diff
@@ -1,65 +1,70 @@
 package main
 
-/*
-#include <stdlib.h>
-*/
-import "C"
-
 import (
-	"fmt"
+	"context"
+	"errors"
 	"time"
-	"unsafe"
 
-	"github.com/calyptia/plugin/input"
+	"github.com/calyptia/plugin"
+	"github.com/calyptia/cmetrics-go"
 )
 
-//export FLBPluginRegister
-func FLBPluginRegister(def unsafe.Pointer) int {
-	return input.FLBPluginRegister(def, "gdummy", "dummy GO!")
-}
+var counter_success *cmetrics.Counter
+var counter_failure *cmetrics.Counter
 
-//export FLBPluginInit
-// (fluentbit will call this)
-// plugin (context) pointer to fluentbit context (state/ c code)
-func FLBPluginInit(plugin unsafe.Pointer) int {
-	// Example to retrieve an optional configuration parameter
-	param := input.FLBPluginConfigKey(plugin, "param")
-	fmt.Printf("[flb-go] plugin parameter = '%s'\n", param)
-	return input.FLB_OK
+func init() {
+	plugin.RegisterInput("gdummy", "dummy GO!", gdummyPlugin{})
 }
 
-//export FLBPluginInputCallback
-func FLBPluginInputCallback(data *unsafe.Pointer, size *C.size_t) int {
-	now := time.Now()
-	flb_time := input.FLBTime{now}
-	message := map[string]string{"message": "dummy"}
+type gdummyPlugin struct{}
 
-	entry := []interface{}{flb_time, message}
+func (plug gdummyPlugin) Init(ctx context.Context, conf plugin.ConfigLoader, cmt *cmetrics.Context) error {
+	var err error
 
-	enc := input.NewEncoder()
-	packed, err := enc.Encode(entry)
+	counter_success, err = cmt.CounterCreate("fluentbit", "input",
+		"operation_succeeded_total", "Total number of succeeded operations", []string{"plugin_name"})
 	if err != nil {
-		fmt.Println("Can't convert to msgpack:", message, err)
-		return input.FLB_ERROR
+		return errors.New("Cannot create counter_success")
 	}
 
-	length := len(packed)
-	*data = C.CBytes(packed)
-	*size = C.size_t(length)
-	// For emitting interval adjustment.
-	time.Sleep(1000 * time.Millisecond)
+	counter_failure, err = cmt.CounterCreate("fluentbit", "input",
+		"operation_failed_total", "Total number of failed operations", []string{"plugin_name"})
+	if err != nil {
+		return errors.New("Cannot create counter_failure")
+	}
 
-	return input.FLB_OK
+	return nil
 }
 
-//export FLBPluginInputCleanupCallback
-func FLBPluginInputCleanupCallback(data unsafe.Pointer) int {
-	return input.FLB_OK
-}
+func (plug gdummyPlugin) Collect(ctx context.Context, ch chan<- plugin.Message) error {
+	tick := time.NewTicker(time.Second)
+
+	for {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			if err != nil && !errors.Is(err, context.Canceled) {
+				err = counter_failure.Inc(time.Now(), []string{"in_gdummy"})
+				if err != nil {
+					return err
+				}
+				return err
```

Comment:
if u will return the `err` anyway so maybe u can do this:
```suggestion
				return counter_failure.Inc(time.Now(), []string{"in_gdummy"})
```

Reply:
Ah, it seems good!